lib/expired_record_cleaner.rb

```diff
@@ -0,0 +1,53 @@
+class ExpiredRecordCleaner
+  attr_reader :verbose, :fiscal_year_start
+
+  def initialize(dtim, ok_to_act = false, verbose = false)
+    @fiscal_year_start = self.class.fiscal_year_start(dtim)
+    @ok_to_act = ok_to_act
+    @verbose = verbose
+  end
+
+  def vacuum_old_proposals
+    proposals = Proposal.pending.where("created_at < '#{@fiscal_year_start}'")
+    ids = []
+    proposals.each do |proposal|
```

Comment:
what about doing `map` here and not worrying about returning the ids? Since we are already outputting the `"#{proposal.public_id} -> #{proposal.requester.email_address}"` below, seems like we have some level of logs documenting what was deleted.